core/src/main/java/com/linecorp/armeria/common/unsafe/PooledHttpRequest.java

```diff
@@ -0,0 +1,119 @@
+     * the trailers of the response are received fully.
+     */
+    default CompletableFuture<PooledAggregatedHttpRequest> aggregateWithPooledObjects() {
+        return aggregateWithPooledObjects(defaultSubscriberExecutor());
+    }
+
+    /**
+     * Aggregates this request. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the request are received fully.
+     */
+    default CompletableFuture<PooledAggregatedHttpRequest> aggregateWithPooledObjects(ByteBufAllocator alloc) {
+        requireNonNull(alloc);
+        return aggregateWithPooledObjects(defaultSubscriberExecutor(), alloc);
+    }
+
+    /**
+     * Aggregates this request. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the request are received fully.
+     */
+    default CompletableFuture<PooledAggregatedHttpRequest> aggregateWithPooledObjects(EventExecutor executor) {
+        requireNonNull(executor);
+        return aggregateWithPooledObjects(executor, PooledByteBufAllocator.DEFAULT);
+    }
+
+    /**
+     * Aggregates this request. The returned {@link CompletableFuture} will be notified when the content and
+     * the trailers of the request are received fully.
+     */
+    default CompletableFuture<PooledAggregatedHttpRequest> aggregateWithPooledObjects(
+            EventExecutor executor, ByteBufAllocator alloc) {
+        final CompletableFuture<AggregatedHttpRequest> future = new EventLoopCheckingFuture<>();
+        final HttpRequestAggregator aggregator = new HttpRequestAggregator(this, future, alloc);
+        subscribe(aggregator, executor, SubscriptionOption.WITH_POOLED_OBJECTS);
+        return future.thenApply(DefaultPooledAggregatedHttpRequest::new);
+    }
+
+    /**
+     * Aggregate this request without pooled objects. When operating on {@link PooledHttpRequest}, this should
+     * be avoided.
+     *
+     * @deprecated Use {@link #aggregateWithPooledObjects()}.
+     */
+    @Override
+    @Deprecated
+    default CompletableFuture<AggregatedHttpRequest> aggregate() {
+        return HttpRequest.super.aggregate();
+    }
+
+    /**
+     * Aggregate this request without pooled objects. When operating on {@link PooledHttpRequest}, this should
+     * be avoided.
+     *
+     * @deprecated Use {@link #aggregateWithPooledObjects(EventExecutor)}.
+     */
+    @Override
+    @Deprecated
+    default CompletableFuture<AggregatedHttpRequest> aggregate(EventExecutor executor) {
+        return HttpRequest.super.aggregate(executor);
+    }
+
+    /**
+     * Converts this {@link PooledHttpRequest} to an unpooled {@link HttpRequest}. Only one of this
+     * {@link PooledHttpRequest} or the returned {@link HttpRequest} can be subscribed to, if both are attempted
+     * to be used, it will cause an error or bad behavior.
+     */
+    HttpRequest toUnpooled();
```

Comment:
`toUnpooled` or `asUnpooled`?

Reply:
We have others like `toHttpResponse` and I think I followed that. Not sure if either is better 😅